bgpd/bgp_attr.c

```diff
@@ -3618,7 +3618,10 @@ enum bgp_attr_parse_ret bgp_attr_parse(struct peer *peer, struct attr *attr,
 				 * a stack buffer, since they perform bounds checking
 				 * and we are working with untrusted data.
 				 */
-				unsigned char ndata[peer->max_packet_size];
+				uint32_t max_packet_size =
+					atomic_load_explicit(&peer->max_packet_size,
+							     memory_order_relaxed);
```

Comment:
So I think this may mess with the data cache and so could be rather expensive as a result. Can we instead simply guarantee that peer's are allocated with the required alignment to guarantee `(offsetof(struct peer, max_packet_size) % 4) == 0`?

Reply:
Hmm `max_packet_size` is actually a `uin16_t`. This is probably a threading error I guess, and not an alignment one. Can you try changing `max_packet_size` to `int` rather than using the atomic function and see if that fixes the error?

Reply:
I fail to see how not using a atomic value is going to prevent two different threads from reading/writing to the same data at the same time.  In any event I'm just removing this one